### ISSUE TITLE
Refactor samples into Gists and submodules

### ISSUE THREAD

**Issue by andymeneely** [andymeneely/squib#117]
Those samples are currently the most important piece of documentation we have - yet they are dumped into the junk drawer of the samples/ folder. Instead, let's move all them out to their own Gists, which are just Git repositories anyway. We can then hook the original repo into it via Git submodules.

Also, this makes it easier for to contribute their own Gists too!

I would also like to incorporate the `spec/samples/expected` folder into each Gist. That way people can see the code and the expected output side-by-side.

**Comment by andymeneely** [comment]
Ok this effort is coming together nicely I think. Here's the checklist for samples to convert.

  - [ ] autoscale_font.rb
  - [ ] backend.rb
  - [ ] basic.rb
  - [ ] cairo_access.rb
  - [ ] color_shortcuts.rb
  - [ ] colors.rb
  - [ ] config_text_markup.rb
  - [ ] csv_import.rb
  - [ ] custom_config.rb
  - [ ] draw_shapes.rb
  - [ ] embed_text.rb
  - [ ] excel.rb
  - [ ] gradients.rb
  - [ ] hand.rb
  - [ ] hello_world.rb
  - [ ] layouts.rb
  - [ ] layouts_builtin.rb
  - [ ] load_images.rb
  - [ ] portrait  -ndscape.rb
  - [ ] ranges.rb
  - [ ] saves.rb
  - [ ] showcase.rb
  - [ ] text_options.rb
  - [ ] tgc_proofs.rb
  - [ ] unicode.rb
  - [ ] units.rb

**Comment by andymeneely** [comment]
I started doing this, but in retrospect, I ended up hating this choice. Submodules make sense, but they interrupt even the most basic Git workflow for me. It's not a pleasant solution.

I'm actually gonna start reversing this set of changes and start keeping everything in the main repo. The benefits of my samples being in separate repos is not really clear - I can just embed them on the readthedocs site anyway.
